message_app_server/message_app/model.py

```diff
@@ -64,8 +64,12 @@ def delete(cls, username: str) -> Union[User, None]:
         return user
 
     @classmethod
-    def select(cls, username: str) -> Union[User, None]:
-        user = User.query.filter_by(username=username).first()
+    def select(cls, username="*", user_id="*", email="*") -> Union[User, None]:
+        user = User.query.filter(and_(
+                User.username == username if username != "*" else true(),
+                User.id == user_id if user_id != "*" else true(),
+                User.email == email if email != "*" else true()
```

Comment:
So here we reason that a user will be uniquely identified with tuple `(username, id, email)` ?

Reply:
Yes, but we only need to pass one of them to get the user from the database.

Reply:
I do believe that the only thing that we should use is the ID, because we may allow users to change their email and probably username.

Reply:
We let users log in with their username or email address, so we need to query with username and email address too. Also, I won't think using any of them will cause trouble because all three columns are unique.